java/src/main/java/com/google/oak/util/BUILD

```diff
@@ -25,3 +25,9 @@ java_library(
     name = "util",
     srcs = glob(["*.java"]),
 )
+
+java_library(
+    name = "util_android",
+    srcs = glob(["*.java"]),
+    constraints = ["android"],
```

Comment:
Some targets (not this one) are forced to split rules since their downstream deps, like grpc, have split rules. Otherwise I added splits as future-proofing -- otherwise if/when we need to add a split dep to util_android in the future, we'd need to go back and update all the targets that depend on it.

I don't know how much some of these targets fluctuate so if you think the scenario above is unlikely, I'm happy to reduce the splits as much as possible.

Reply:
I looked into it further and constraints is no longer supported in bazel (github.com/bazelbuild/bazel/issues/3555). This means a single target can't serve both java and android and we need to maintain two build trees after all.